src/rules/noUnnecessaryClassRule.ts

```diff
@@ -0,0 +1,149 @@
+            * \`"allow-static-only"\` ignores classes whose members are static.`,
+        options: {
+            type: "array",
+            items: {
+                type: "string",
+            },
+            minLength: 0,
+            maxLength: 3,
+        },
+        optionExamples: [true, ["allow-empty-class", "allow-constructor-only"]],
+        type: "functionality",
+        typescriptOnly: false,
+    };
+    /* tslint:enable:object-literal-sort-keys */
+
+    public static FAILURE_CONSTRUCTOR_ONLY = "Every member of this class is a constructor. Use functions instead.";
+    public static FAILURE_STATIC_ONLY = "Every member of this class is static. Use namespaces or plain objects instead.";
+    public static FAILURE_EMPTY_CLASS = "This class has no members.";
+
+    public apply(sourceFile: ts.SourceFile): Lint.RuleFailure[] {
+        return this.applyWithWalker(new NoUnnecessaryClassWalker(sourceFile, this.ruleName, this.ruleArguments));
+    }
+}
+
+const OPTION__ALLOW_CONSTRUCTOR_ONLY = "allow-constructor-only";
+const OPTION__ALLOW_EMPTY_CLASS = "allow-empty-class";
+const OPTION__ALLOW_STATIC_ONLY = "allow-static-only";
+
+class NoUnnecessaryClassWalker extends Lint.AbstractWalker<string[]> {
+    public walk(sourceFile: ts.SourceFile) {
+        const checkIfUnnecessaryClass = (node: ts.Node): void => {
+            if (isClassDeclaration(node) && !hasExtendsClause(node)) {
+                return this.checkMembers(node, checkIfUnnecessaryClass);
+            }
+            return ts.forEachChild(node, checkIfUnnecessaryClass);
+        };
+        ts.forEachChild(sourceFile, checkIfUnnecessaryClass);
+    }
+
+    private checkMembers(node: ts.ClassDeclaration, checkIfUnnecessaryClass: (node: ts.Node) => void) {
+        if (node.members.length === 0) {
+            if (!this.hasOption(OPTION__ALLOW_EMPTY_CLASS)) {
+                this.addFailureAtNode(getChildOfKind(node, ts.SyntaxKind.ClassKeyword)!, Rule.FAILURE_EMPTY_CLASS);
+            }
+            return;
+        }
+
+        const allMembersAreConstructors = node.members.every(isConstructorDeclaration);
+        if (
+            allMembersAreConstructors &&
+            !this.hasOption(OPTION__ALLOW_CONSTRUCTOR_ONLY) &&
+            !node.members.some(isConstructorWithShorthandProps)
+        ) {
+            this.addFailureAtNode(
+                getChildOfKind(node, ts.SyntaxKind.ClassKeyword, this.sourceFile)!,
+                Rule.FAILURE_CONSTRUCTOR_ONLY,
+            );
+        }
+
+        if (
+            !allMembersAreConstructors &&
+            !this.hasOption(OPTION__ALLOW_STATIC_ONLY) &&
+            !node.members.some(isNonStaticMember)
+        ) {
+            this.addFailureAtNode(
+                getChildOfKind(node, ts.SyntaxKind.ClassKeyword, this.sourceFile)!,
+                Rule.FAILURE_STATIC_ONLY,
+            );
+        }
+
+        /* Check for classes nested in constructors */
+        for (const member of node.members) {
```

Comment:
you can remove this block if you remove the return at the call site (see my other comment above)